### ISSUE TITLE
Question about Transcode Videos on Google Cloud Storage

### ISSUE THREAD

**Issue by rlee1990** [GoogleCloudPlatform/firebase-extensions#6]
How are we able to play the videos after they are transcoded. I do not see a master.meu8 file I saw a manifest.m3u8 file but it is not playable.

**Comment by i14h** [comment]
Thanks for reporting this @rlee1990 
@Salakar @ZackAkil can you please look into this?

**Comment by rlee1990** [comment]
@Salakar or @ZackAkil is there any progress on this at all?

**Comment by rlee1990** [comment]
@i14h is there any progress on this? I would like to be able to use this with an app I am working on but this is holding it up.

**Comment by i14h** [comment]
@rlee1990 yes, stay tuned. We'll update this thread in a couple of days.

**Comment by rlee1990** [comment]
Thanks

**Comment by dackers86** [comment]
Hi @rlee1990.

The output it generated using the `transcode api`. The full list of outputs for this extension are listed here https://cloud.google.com/transcoder/docs/concepts/overview#supported_inputs_outputs

There is a `.mp4` if this could be what you are looking for?

This can be downloaded, for example `gsutil cp gs://my-bucket/my-file-path/hd.mp4 .`

**Comment by rlee1990** [comment]
@dackers86 No when any video is transcoded they way you doing it a m3u8 file is made no weather that is called master or manifest don’t think matters but what is in the file does. It should hold all the information like a playlist of sorts. The one that was made is missing paths to the mp4 files.

**Comment by dackers86** [comment]
@rlee1990 I believe the different output is based on the Template Id that defines the `job configuration`.

By default , we use `preset/web-hd`. I cannot see an example configuration that has a master.m3u8? Could creating your own example per https://cloud.google.com/transcoder/docs/concepts/config-examples be a potential solution?

**Comment by rlee1990** [comment]
@dackers86 Forget the name master the issue is the file that is returned that manages what gets played does not point to any of the videos correctly. The transcoding creates a SD and an HD video out put along with a manifest file but the data in that file is not configured to point to the video files like it should and won't play in any player like it should.

**Comment by dackers86** [comment]
Yes, apologies I do understand that. The issue here in terms of this extension is that under the hood, it simply calls the `transcoder api`. 

The parameters for the file input and out `uri` are provided along with the necessary configuration. The Node client https://googleapis.dev/nodejs/transcoder/latest/index.html then takes over to generate the files in the directory provided.

If this is an issue that cannot be fixed a custom template, perhaps we could create an issue with above library as any missing data would generated at that location.

**Comment by rlee1990** [comment]
thanks.

**Comment by dackers86** [comment]
Hi All.

From the closed related issues, I am assuming that this has now resolved? Closing for now, but feel free to ping if this needs re-opening.
